packages/babel-generator/src/generators/types.js

```diff
@@ -106,6 +106,67 @@ export function ArrayExpression(node: Object) {
 
 export { ArrayExpression as ArrayPattern };
 
+export function RecordExpression(node: Object) {
+  const props = node.properties;
+
+  let startToken;
+  let endToken;
+  if (this.format.recordAndTupleSyntaxType === "bar") {
+    startToken = "{|";
+    endToken = "|}";
+  } else if (this.format.recordAndTupleSyntaxType === "hash") {
+    startToken = "#{";
+    endToken = "}";
+  } else {
+    throw new Error(
+      `${this.format.recordAndTupleSyntaxType} is not a valid recordAndTuple syntax type`,
```

Comment:
This error message should mention the option name, and also be easy to understand when the option is not provided. WDYT about this?

```js
      `The "recordAndTupleSyntaxType" generator option must be either "bar" or "hash" (${JSON.stringify(this.format.recordAndTupleSyntaxType)} received).`,
```

Also, please add a test for an invalid value of the option and for when the option is not set.

Reply:
@nicolo-ribaudo pushed a change to include this new message, and added the tests you mentioned.

The `babel-generator` test runner doesn't support the standard `throws` option in `options.json` (looks like it isn't using the same infra as everything else). I took the liberty of adding support for `throws` to the package's test suite, let me know if this should exist somewhere else.